### ISSUE TITLE
Add new index.html

### ISSUE THREAD

**Issue by paulvickers** [mcolburn/durham#77]
The previous one turned out to have some unmatched tags. Sorry! Can we try this version please?

**Comment by mcolburn** [comment]
I pasted the index.html content into this online html validator:

https://validator.w3.org￼

I fixed all error it found, and pushed them to https://github.com/mcolburn/durham

So, you need to do a pull.

Even after fixing all errors found by the validator, the generator still hangs when I use the Durham Divine Liturgy html.

The editor works, though.

Here is what I think we need to do:

Presbytera can use OLW to enter the remaining Romanian.
I will go ahead and get you set up to use Doxa.  You can use it to generate the PDF file.

But, keep in mind that Doxa is still under development.

How does that sound to you?

**Comment by paulvickers** [comment]
Hmmm. How much control will Doxa give me over layout (fonts, responses next to/below actors, cover pages, etc.)? Might it be worth reverting back to the index.html that did work?

You will also see from the email I sent that there is a problem with the Romanian text entry. The template seems to be putting en_UK_durlash keys into the Romanian version rather than ro_UK_dur.

**Comment by mcolburn** [comment]
Doxa will give you total control over the layout, the fonts, etc.

But, let me think about the issues you have been running into and whether there is a better temporary alternative for you to use for now, e.g. using a pure OSLW solution.

**Comment by mcolburn** [comment]
I pushed the latest index.html.  The editor seems to be OK.  Please verify.  However, the generator still hangs.  I will see if I can find an obvious reason.

**Comment by paulvickers** [comment]
Thanks, Michael. I pushed one more up 20 mins or so ago which had a minor change.

**Comment by mcolburn** [comment]
I pushed to OLW.  I am trying to get a debugger going on my dev for OLW to see if I can spot an obvious reason for why the generator hangs.  Intellij Idea's debugger has stopped working on my code and can't figure out why.  I just now got Visual Studio Code to work debugging java.  So, perhaps in a few hours I will know what the problem is.

**Comment by paulvickers** [comment]
I just spotted one empty (of text) key pair in the index.html which I have now removed (just below the heading of the Holy Anaphora):
````
<tr>
<td class='leftCell'><span class='kvp' data-key='properties_gr_GR_cog|media.key'><span class='media-group-empty'></span></span>
</td>
<td class='rightCell'><span class='kvp' data-key='properties_en_US_goa|media.key'><span class='media-group-empty'></span></span> </td>
</tr>
````

**Comment by mcolburn** [comment]
Empty is not a problem.

**Comment by paulvickers** [comment]
ok, just that in the editor it shows up as `properties ~ media.key`

**Comment by mcolburn** [comment]
Yes, that is a place holder so the user has something to click on to open the record editor.  It displays like that if the value is empty.

If it is not needed for your use, it can be removed as you did.

**Comment by paulvickers** [comment]
Ah, I see. 

Is it worth me replacing all the uk_durlash domain tags with us_goa do you think?

**Comment by mcolburn** [comment]
Hold off for now.

I finally have breakpoints set in the correct place in the OLW code and I will be able to determine where and hopefully why the generation is hanging, without reporting an error back.

**Comment by mcolburn** [comment]
Making progress.

I now know the exact line in the code where an array out of bounds error is thrown.  Don't know the cause yet, or why OLW doesn't report the error to the user....

**Comment by paulvickers** [comment]
Weird that it should suddenly begin doing that. It couldn't be anything to do with large comment blocks, could it?

**Comment by paulvickers** [comment]
Might it be worth putting back in the last version of index.html that didn't throw an error? If it works, I could then clean that one up to remove the mismatched tags & domains.

**Comment by mcolburn** [comment]
Paul, please test on OLW.  I found the source of the array out of bounds error in my code.  It was an execution path caused when a data-key did not parse properly.

The durham liturgy works now for both the editor and generator.

I need to work with the data from the index.html file to figure out why some of the data-key values are not being parsed properly.  Even though the output seems correct and OLW is happy, I am not until I get to the bottom of the instances where the data-key did not parse correctly.

**Comment by paulvickers** [comment]
Bless you, Michael. I hope this parsing issue helps with Doxa too...
I still need to make a couple more tweaks to the template (there are some ChoirL and ChoirR instances that I just want to be Choir) so will upload a new version of index.html later. I hope this won't cause any further issues for you.

**Comment by mcolburn** [comment]
Hi, Paul!

No worries!

Glad it is working correctly now.

Even though the array out of bounds error was caught in the OLW server code and passed in the http response back to the OLW front-end app (running on the browser), in this case, I did not check the status code.  I only checked to see if I had received data.  So, the response was received, but not handled properly if there was no data in the response.  That left the spinner (wait) icon in place indefinitely.  Yesterday, I fixed it so the code checks for a 200 (ok) response code and displays a message to the user if there is a problem and removes the spinner icon.

In Doxa, I always check the response status code.

-Michael

**Comment by paulvickers** [comment]
Thanks again. I've pushed what I hope is the final index.html to the repo. I decided to replace all Choir IDs with People IDs.

**Comment by mcolburn** [comment]
Pushed at 16:05 your time.  

It usually takes 2-5 minutes to be available to OLW.  The push I make triggers netlify to republish the site accessed at https://books.liml.org.  OLW reads it, as well as the GOA DCS, to list available books and services.

**Comment by paulvickers** [comment]
Perfect, thanks! I'll close this issue now.
